druid-mr/src/main/java/com/yahoo/druid/hadoop/DruidHelper.java

```diff
@@ -40,23 +36,16 @@
   public List<String> getSegmentPathsToLoad(final String dataSource, final Interval interval,
       final String storageDir, final String overlordUrl) {
     List<DataSegment> segmentsToLoad = getSegmentsToLoad(dataSource, interval, overlordUrl);
-    VersionedIntervalTimeline<String, DataSegment> timeline = new VersionedIntervalTimeline<String, DataSegment>(
-          Ordering.<String>natural().nullsFirst()
-      );
-    for (DataSegment segment : segmentsToLoad) {
-      timeline.add(segment.getInterval(), segment.getVersion(), segment.getShardSpec().createChunk(segment));
-    }
-    final List<TimelineObjectHolder<String, DataSegment>> timeLineSegments = timeline.lookup(interval);
```

Comment:
@himanshug  I wonder if this logic is necessary, as Druid already do it by [here] (https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java#L121-L135)